libraries/iio/iio_types.h

```diff
@@ -117,6 +117,12 @@ struct iio_channel {
 	char			*name;
 	/** Chanel type */
 	enum iio_chan_type	ch_type;
+	/** Channel number when the same channel type */
+	int 			channel;
```

Comment:
Nitpick: I think it makes more sense if you add this channel in the 3-rd commit and  channel2 in the 4-th commit.